code/datums/spells/ethereal_jaunt.dm

```diff
@@ -15,44 +15,77 @@
 
 	var/jaunt_duration = 50 //in deciseconds
 	var/jaunt_in_time = 5
+
+	///Do we do the glowy wizard thing on the tile the wizard entered / exit
+	var/has_jaunt_effect = TRUE
 	var/jaunt_in_type = /obj/effect/temp_visual/wizard
 	var/jaunt_out_type = /obj/effect/temp_visual/wizard/out
+	/// Do we show the fun blue water effect spread out thing
+	var/has_smoke_jaunt_effect = TRUE
+
+	/// Sound to play on jaunting
+	var/jaunt_enter_sound = 'sound/magic/ethereal_enter.ogg'
+
+	/// Sound to play when we start exiting jaunt
+	var/jaunt_exit_sound = 'sound/magic/ethereal_exit.ogg'
+
+	/// Do we unstun (currently shadowling)
+	var/unstuns = FALSE
+
+	/// Do we play a message when activating / deactivating the spell, in the format of "<span class='warning'>(user's name) (message to be displayed)</span>"
+	var/plays_message = FALSE
+	var/message_in = "Should not"
+	var/message_out = "See this"
+
+	/// Do we pop out instantly when the spell ends (not reccomended if using jaunt effects)
+	var/instant_pop_out = FALSE
 
 	action_icon_state = "jaunt"
 
 /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/cast(list/targets, mob/user = usr) //magnets, so mostly hardcoded
-	playsound(get_turf(user), 'sound/magic/ethereal_enter.ogg', 50, 1, -1)
+	playsound(get_turf(user), jaunt_enter_sound, 50, 1, -1)
 	for(var/mob/living/target in targets)
 		if(!target.can_safely_leave_loc()) // No more brainmobs hopping out of their brains
 			to_chat(target, "<span class='warning'>You are somehow too bound to your current location to abandon it.</span>")
 			continue
 		INVOKE_ASYNC(src, .proc/do_jaunt, target)
 
 /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/proc/do_jaunt(mob/living/target)
+	if(plays_message)
+		target.visible_message("<span class='warning'>[target] [message_in]</span>")
```

Comment:
I think the values and placement of `message_in` and `message_out` in this proc should be swapped. First the user jaunts *out*, then toward the end of the proc, the user jaunts back *in*.
```suggestion
		target.visible_message("<span class='warning'>[target] [message_out]</span>")
```

Reply:
Keeping as is, due to the way soundfiles / logic was before / still is, as the user enters jaunt ('sound/magic/ethereal_enter.ogg') then exits jaunt later ('sound/magic/ethereal_exit.ogg'). The sound / naming is for entering jaunt, not jaunting out.